cli/lsp/diagnostics.rs

```diff
@@ -217,6 +217,7 @@ impl DiagnosticsServer {
                     snapshot.clone(),
                     &config,
                     &ts_server,
+                    token.clone(),
```

Comment:
With this change we might want to consider removing or reducing the 200ms delay on line 208.

Reply:
there is also antidotal evidence too by using a debug build and editing a large project in vscode... I can try to take it for a spin later... though if we drop that, we should do it as a follow up PR, I suspect.

Reply:
That's a good idea! Though I just tested it now and removing the delay is still much slower ☹

Before (this PR):

```
Starting Deno benchmark
-> Start benchmarking lsp
   - Simple Startup/Shutdown
      (10 runs, mean: 508ms)
   - Big Document/Several Edits 
      (5 runs, mean: 1879ms)
   - Find/Replace
      (10 runs, mean: 78ms)
   - Code Lens
      (10 runs, mean: 612ms)
<- End benchmarking lsp
```

After (debounce removed):

```
Starting Deno benchmark
-> Start benchmarking lsp
   - Simple Startup/Shutdown 
      (10 runs, mean: 295ms)
   - Big Document/Several Edits 
      (5 runs, mean: 4668ms)
   - Find/Replace
      (10 runs, mean: 78ms)
   - Code Lens
      (10 runs, mean: 402ms)
<- End benchmarking lsp
```

Reply:
The test might not be triggering it, because I don't think it is sending and setting cancellation tokens. I think we would need to modify the test, based on observed messages to and from the server.